daemon/src/projection.rs

```diff
@@ -133,21 +133,21 @@ impl State {
         let _ = std::mem::replace(&mut self.cfds, cfds);
     }
 
-    fn amend_cfd_proposal(
-        &mut self,
-        order: OrderId,
-        proposal: Option<UpdateCfdProposal>,
-    ) -> Result<()> {
+    fn amend_cfd_proposal(&mut self, order: OrderId, proposal: Option<UpdateCfdProposal>) {
         if let Some(proposal) = proposal {
             self.proposals.insert(order, proposal);
             tracing::trace!(%order, "Cfd proposal got updated");
-        } else {
-            if self.proposals.remove(&order).is_none() {
-                anyhow::bail!("Could not find proposal with order id: {}", &order)
-            }
-            tracing::trace!(%order, "Removed cfd proposal");
-        };
-        Ok(())
+
+            return;
+        }
+
+        if self.proposals.remove(&order).is_none() {
+            tracing::trace!(%order, "Cannot remove cfd proposal: unknown");
```

Comment:
Good change - the caller doesn't really care if the other part of the system misbehaved as it can't fix it anyway.

One thing though:
perhaps this should be at least `debug`, or even `error`? I'm a bit worried that if we bury this log message too deep we will not notice if our logic is flawed. Ideally, this should never happen.

Reply:
Can make it debug yeah.

`error` I would only use if the answer to "Would I want an automated monitoring tool to wake me up at 3am if this happens?" is yes 😁

Reply:
debug sounds good 👍